store/precomputed_key/weaveVM/wvm.go

```diff
@@ -0,0 +1,212 @@
+package weavevm
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"github.com/Layr-Labs/eigenda-proxy/common"
+	rpc "github.com/Layr-Labs/eigenda-proxy/store/precomputed_key/weavevm/rpc"
+	signer "github.com/Layr-Labs/eigenda-proxy/store/precomputed_key/weavevm/signer"
+	weaveVMtypes "github.com/Layr-Labs/eigenda-proxy/store/precomputed_key/weavevm/types"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	cache "github.com/patrickmn/go-cache"
+)
+
+type WeaveVM interface {
+	SendTransaction(ctx context.Context, to string, data []byte) (string, error)
+}
+
+// Store...wraps weaveVM client, ethclient and concurrent internal cache
+type Store struct {
+	weaveVMClient WeaveVM
+	log           log.Logger
+	txCache       *cache.Cache
```

Comment:
what's the purpose of this cache? Does it change hardware requirements and could it lead to OOM exceptions with high usage?

Reply:
right now we use it to store a mapping between commitment key and weave vm transaction hash
it's only temporary solution and should be changed alongside with arbitrary blob sizes
with rough estimations of 1 commitment per second and 15 days TTL + some overhead this cache shouldn't be more than 145MB
I tried to pick the most simple and "essentials" implementation which is thread-safety, has ttl expiry ,tested